vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -452,19 +461,60 @@ public boolean isInvalid() {
         return isInvalidBoolean();
     }
 
+    private ValidationResult checkValidity(LocalDate value) {
+        ValidationResult greaterThanMax = checkGreaterThanMax(value, max);
+        if (greaterThanMax.isError()) {
+            return greaterThanMax;
+        }
+
+        ValidationResult smallerThanMin = checkSmallerThanMin(value, min);
+        if (smallerThanMin.isError()) {
+            return smallerThanMin;
+        }
+
+        return ValidationResult.ok();
+    }
+
+    public static <V extends Comparable<V>> ValidationResult checkGreaterThanMax(
```

Comment:
Let's make these methods private and non-static as they aren't part of public API.

Reply:
Let's discuss with @TatuLund if it's okay to introduce something like a `component-base` package after all.